arangod/Statistics/StatisticsWorker.cpp

```diff
@@ -1131,6 +1131,12 @@ void StatisticsWorker::run() {
 
   uint64_t seconds = 0;
   while (!isStopping()) {
+    TRI_IF_FAILURE("StatisticsWorker::bypass") {
+      CONDITION_LOCKER(guard, _cv);
+      guard.wait(1000 * 1000);
+      continue;
+    }
```

Comment:
Can you explain this for me?

Reply:
Yes, this is a failure condition we can trigger (only) in tests. If enabled, it will make the StatisticsWorker thread do nothing. Just make it sleep for a while, but still react to potential server shutdown. The failure point gets enabled in a test that should not be interrupted by background statistics writing.

Reply:
Thanks